util/makeROSCompatibleURDF.m

```diff
@@ -0,0 +1,65 @@
+function makeROSCompatibleURDF(input_urdf_filename,output_urdf_filename)
+% Drake supports urdf input beyond what ROS and other URDF readers
+% support.  Additional drake xml tags will simply be ignored by other URDF
+% parsers, but parameters and the ability to write matlab expressions in
+% place of floats can cause problems.  This script will parse the
+% parameters and write a new urdf which is compliant.
+%
+% @param input_urdf_filename string containing the filename
+% @param output_urdf_filename optional string containing the output
+% filename @default: input_urdf_filename with '_compatible' inserted in the name
+% just before the urdf
+
+if nargin<2,
+  [~,~,ext] = fileparts(input_urdf_filename);
+  output_urdf_filename = strrep(input_urdf_filename,ext,['_compatible',ext]);
+end
+
+param_db = struct();
+urdf = xmlread(input_urdf_filename);
+
+% parse parameters
+parameters = urdf.getElementsByTagName('parameter');
+for i=0:(parameters.getLength()-1)
+  param_db = parseParameter(param_db,parameters.item(i));
+end
+
+% now iterate through ALL nodes and evaluate them based on their parameters
+nodes = urdf.getElementsByTagName('*');
+for i=0:(nodes.getLength()-1)
+  attributes = nodes.item(i).getAttributes();
+  for j=0:(attributes.getLength()-1)
+    att = attributes.item(j);
+    att.setValue(parseParamString(param_db,char(att.getValue())));
+  end  
+end
+
+xmlwrite(output_urdf_filename,urdf);
+fprintf('Output written to %s\n',output_urdf_filename);
+
+end
+
+
+function param_db = parseParameter(param_db,node)
```

Comment:
unfortunately, it's behavior is a sufficiently different that I think duplication+modification is probably the right way to go.

On Feb 2, 2015, at 5:07 PM, Andres Valenzuela <notifications@github.com> wrote:

> In util/makeROSCompatibleURDF.m:
> 
> > +nodes = urdf.getElementsByTagName('*');
> > +for i=0:(nodes.getLength()-1)
> > +  attributes = nodes.item(i).getAttributes();
> > +  for j=0:(attributes.getLength()-1)
> > +    att = attributes.item(j);
> > +    att.setValue(parseParamString(param_db,char(att.getValue())));
> > +  end  
> > +end
> > +
> > +xmlwrite(output_urdf_filename,urdf);
> > +fprintf('Output written to %s\n',output_urdf_filename);
> > +
> > +end
> > +
> > +
> > +function param_db = parseParameter(param_db,node)
> Is there a way we could avoid duplicating this function between RBM and here?
> 
> —
> Reply to this email directly or view it on GitHub.
>